tests/phpunit/api/v3/ChContribution/CreateTest.php

```diff
@@ -26,24 +37,121 @@ public function setUpHeadless() {
+      'option_group_id' => 'ch_fund',
+      'value' => 'CH+99999',
+    ]);
+    $chFundMap = $this->callAPISuccess('OptionValueCH', 'get', ['value' => 'CH+99999']);
+    $contact = $this->individualCreate();
+    $this->callAPISuccess('OptionValueCH', 'create', ['financial_type_id' => $this->fund['id'], 'id' => $chFundMap['id']]);
+    $contribution = $this->callAPISuccess('CHContribution', 'create', [
+      'contact_id' => $contact,
+      'ch_fund' => 'CH+99999',
+      'payment_instrument_id' => 'Credit Card',
+      'total_amount' => '100',
+    ]);
+    $getResult = $this->callAPISuccess('Contribution', 'get', ['return' => ['custom_' . $this->customField['id']], 'id' => $contribution['id']]);
+    // Confirm that the custom field has been correctly populated and that the correct financial type was assigned.
+    $this->assertEquals($this->fund['id'], $contribution['values'][$contribution['id']]['financial_type_id']);
+    $this->assertEquals('CH+99999', $getResult['values'][$getResult['id']]['custom_' . $this->customField['id']]);
+    $this->callAPISuccess('Contribution', 'delete', ['id' => $contribution['id']]);
+    $this->callAPISuccess('OptionValue', 'delete', ['id' => $chFund['id']]);
+  }
+
+  public function testUpdatingContributionsWhenChFundMappingChanges() {
+    $chFund = $this->callAPISuccess('OptionValue', 'create', [
+      'label' => 'Test Created CH Fund 1',
+      'option_group_id' => 'ch_fund',
+      'value' => 'CH+99999',
+    ]);
+    $chFundMap = $this->callAPISuccess('OptionValueCH', 'get', ['value' => 'CH+99999']);
+    $contact = $this->individualCreate();
+    $this->callAPISuccess('OptionValueCH', 'create', ['financial_type_id' => $this->fund['id'], 'id' => $chFundMap['id']]);
+    $contribution = $this->callAPISuccess('CHContribution', 'create', [
+      'contact_id' => $contact,
+      'ch_fund' => 'CH+99999',
+      'payment_instrument_id' => 'Credit Card',
+      'total_amount' => '100',
+    ]);
+    $getResult = $this->callAPISuccess('Contribution', 'get', ['return' => ['custom_' . $this->customField['id']], 'id' => $contribution['id']]);
+    // Confirm that the custom field has been correctly populated and that the correct financial type was assigned.
+    $this->assertEquals($this->fund['id'], $contribution['values'][$contribution['id']]['financial_type_id']);
+    $this->assertEquals('CH+99999', $getResult['values'][$getResult['id']]['custom_' . $this->customField['id']]);
+    $this->callAPISuccess('OptionValueCH', 'create', ['id' => $chFundMap['id'], 'financial_type_id' => $this->fund2['id']]);
+    $getResult = $this->callAPISuccess('Contribution', 'get', ['return' => ['custom_' . $this->customField['id'], 'financial_type_id'], 'id' => $contribution['id']]);
+    // Confirm that nothing has happened yet to the contributions.
+    $this->assertEquals($this->fund['id'], $getResult['values'][$contribution['id']]['financial_type_id']);
+    E::updateCHContribution($this->fund2['id'], 'CH+99999');
```

Comment:
@seamuslee001 can we use CHContribution.create API for this as 

```
api3(CHContribution, create, [
'id' => contributionID,
'ch_fund' => 'CH+99999',
])
```

The reason is, we should rely entirely on custom APIs where it is possible, and thus add more API checks.

Reply:
@monishdeb sure, however in terms of testing the current pathways i couldn't see anything that used CHContribution.create API and both methods in the "live" code use this method https://github.com/CanadaHelps/chfunds/blob/bc47ee848bff3ad96f72bc2ab4ffb02b016cc987/CRM/Chfunds/Form/CHFunds.php#L116 https://github.com/CanadaHelps/chfunds/blob/786540cc6481928f86685af6c67c50b972c6c920/chfunds.php#L296 which adds to a batch rather than an immediate update.  I figured it was prudent to test this code flow

Reply:
@seamuslee001 actually in "live" code E::updateCHContribution() has a special purpose as it doesn't ACTUALLY update the contribution's  FT and CH Fund but only make contribution entities in batch in ```civicrm_ch_contribution_batch``` table [here](https://github.com/CanadaHelps/chfunds/blob/master/CRM/Chfunds/ExtensionUtil.php#L105) (that stores the contribution IDs and the new fund and chfund value). Which is later processed by a cron job [UpdateCHContributions](https://github.com/CanadaHelps/chfunds/blob/master/api/v3/Job/UpdateCHContributions.php) 

In your case, it is a simple update, as we just need to update the Fund of a single contribution and reason why we don't need this expensive function to do this simple job (and also E::updateCHContribution() itself depends on cron job to update the batch of contributions) 

In live code, its true that we are not using this CHContribution.create API but it was earlier implemented for CanadaHelp internal use. Therefore to ensure that this API is working properly we should call this API in our test suite wherever possible.